src/observables/ForkJoinObservable.ts

```diff
@@ -3,41 +3,41 @@ import {Subscriber} from '../Subscriber';
 import {PromiseObservable} from './PromiseObservable';
 import {EmptyObservable} from './EmptyObservable';
 import {isPromise} from '../util/isPromise';
+import {isArray} from '../util/isArray';
 
 export class ForkJoinObservable<T> extends Observable<T> {
-  constructor(private sources: Array<Observable<any> |
-                                    Promise<any> |
-                                    ((...values: Array<any>) => any)>) {
-     super();
+  constructor(private sources: Array<Observable<any> | Promise<any>>,
+              private resultSelector?: (...values: Array<any>) => any) {
+    super();
   }
 
   static create(...sources: Array<Observable<any> |
+                                  Array<Observable<any>> |
                                   Promise<any> |
-                                  ((...values: Array<any>) => any)>)
-                                  : Observable<any> {
-    if (sources === null || sources.length === 0) {
+                                  ((...values: Array<any>) => any)>) : Observable<any> {
```

Comment:
After thinking bit more with updated code, came to think in this case function overload signature does not do much thing, simplified by having single signature does polymorphic.